spotlight/losses.py

```diff
@@ -79,8 +83,8 @@ def bpr_loss(positive_predictions, negative_predictions, mask=None):
        uncertainty in artificial intelligence. AUAI Press, 2009.
     """
 
-    loss = (1.0 - F.sigmoid(positive_predictions -
-                            negative_predictions))
+    loss = (1.0 - F.sigmoid((positive_predictions -
+                             negative_predictions) * weights))
```

Comment:
I think the `weights` term should be outside the `(1.0 - F.sigmoid(...))` term. I don't think we want to ever allow this term to become negative.